communication/mavlink_waypoint_handler.cpp

```diff
@@ -881,31 +874,32 @@ void Mavlink_waypoint_handler::auto_landing_handler()
                 print_util_dbg_print("Cust: descent to small alt");
                 state_.mav_mode_custom &= static_cast<Mav_mode::custom_mode_t>(0xFFFFFFE0);
                 state_.mav_mode_custom |= Mav_mode::CUST_DESCENT_TO_SMALL_ALTITUDE;
-                waypoint_hold_coordinates.waypoint = position_estimation_.local_position;
-                waypoint_hold_coordinates.waypoint.pos[Z] = navigation_.takeoff_altitude/2.0f;
+                waypoint_hold_coordinates.position    = ins_.position_lf();
+                waypoint_hold_coordinates.position[Z] = navigation_.takeoff_altitude / 2.0f;
+                waypoint_hold_coordinates.heading     = coord_conventions_get_yaw(ahrs_.qe);
```

Comment:
I am not sure you want to do that here :)
The heading is suppose to be set properly before hand. Otherwise it is a kind of open loop controller at that step !